modules/tf_cloudbuild_workspace/variables.tf

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "GCP project for Cloud Build triggers, state and log buckets."
+  type        = string
+}
+
+variable "location" {
+  description = "Location for build logs/state bucket"
+  type        = string
+  default     = "us-central1"
+}
+
+variable "cloudbuild_sa" {
+  description = "Custom SA email to be used by the CloudBuild trigger. Defaults to being created if empty."
+  type        = string
+  default     = ""
+}
+
+variable "state_bucket_self_link" {
+  description = "Custom GCS bucket for storing TF state. Defaults to being created if empty."
+  type        = string
+  default     = ""
+}
+
+variable "cloudbuild_sa_roles" {
+  description = "Optional to assign to custom CloudBuild SA. Map of project id to list of roles expected."
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "cloudbuild_plan_filename" {
+  description = "Optional Cloud Build YAML definition used for terraform plan. Defaults to using inline definition."
+  type        = string
+  default     = null
+}
+
+variable "cloudbuild_apply_filename" {
+  description = "Optional Cloud Build YAML definition used for terraform apply. Defaults to using inline definition."
+  type        = string
+  default     = null
+}
+
+variable "cloudbuild_env_vars" {
+  description = "Optional list of environment variables to be used in builds. List of strings of form KEY=VALUE expected."
+  type        = list(string)
+  default     = []
+}
+
+variable "buckets_force_destroy" {
+  description = "When deleting the bucket for storing CloudBuild logs/TF state, this boolean option will delete all contained objects. If false, Terraform will fail to delete buckets which contain objects."
+  type        = bool
+  default     = false
+}
+
+variable "substitutions" {
```

Comment:
if the user is providing the full set of substitution, how should the possible substitutions that are created inside the module
`_TF_SA_EMAIL`, `_STATE_BUCKET_NAME`, and `_LOGS_BUCKET_NAME` be handle?

Reply:
Added these as default substs merged with user provided.